### ISSUE TITLE
do histoical decomposition under dynareOBC

### ISSUE THREAD

**Issue by lulu-mao** [tholden/dynareOBC#38]
Hello everyone

I was trying to do historical decomposition under dynareOBC. so I add the following thing in the mod file.

stoch_simul(periods=1000,noprint,order=1, irf=0);
varobs qh pi R yh;
calib_smoother(datafile=firm_loan_hs,diffuse_filter);
shock_decomposition(parameter_set=calibration,datafile=firm_loan_hs)qh yh;

like what I did under  Dynare, but I got an error 

_**Error: File: dynareOBCTempPostScript.m Line: 2 Column: 24
Incorrect use of '=' operator. To assign a value to a variable, use
'='. To compare values for equality, use '=='.

Error in dynareOBCCore (line 720)
    evalin( 'base', 'dynareOBCTempPostScript;' );

Error in dynareOBCSetup (line 297)
            dynareOBC_ = dynareOBCCore( InputFileName, basevarargin,
            dynareOBC_, @() EnforceRequirementsAndGeneratePath(
            Update, OriginalPath, CurrentFolder, dynareOBCPath,
            InputFileName, varargin{:} ) );

Error in dynareOBC (line 711)
        dynareOBCSetup( OriginalPath, CurrentFolder, dynareOBCPath,
        InputFileName, varargin{:} );**_

following the error instruction, i replace the = to ==, so I replace the adding code to 
stoch_simul(periods=1000,noprint,order=1, irf=0);
varobs qh pi R yh;
calib_smoother(datafile==firm_loan_hs,diffuse_filter);
shock_decomposition(parameter_set==calibration,datafile==firm_loan_hs)qh yh;

but I still get another  error 

Error: File: dynareOBCTempPostScript.m Line: 3 Column: 71
Invalid expression. Check for missing multiplication operator,
missing or unbalanced delimiters, or other syntax error. To
construct matrices, use brackets instead of parentheses.

Error in dynareOBCCore (line 720)
    evalin( 'base', 'dynareOBCTempPostScript;' );

Error in dynareOBCSetup (line 297)
            dynareOBC_ = dynareOBCCore( InputFileName, basevarargin,
            dynareOBC_, @() EnforceRequirementsAndGeneratePath(
            Update, OriginalPath, CurrentFolder, dynareOBCPath,
            InputFileName, varargin{:} ) );

Error in dynareOBC (line 711)
        dynareOBCSetup( OriginalPath, CurrentFolder, dynareOBCPath,
        InputFileName, varargin{:} );

I was thinking do  I  use the right code to do historical decomposition?   I've tried to run dynare filename.mod on Matlab instead of running dynareOBC filename.mod, using this way I can get the figure of historical decomposition.  but I don't think running dynare filename.mod can capture the restriction with max and min equations.  has anyone tried to do historical decomposition under dynareOBC?  many thanks.

**Comment by tholden** [comment]
Doing historical decompositions with DynareOBC is not possible at present. Additionally, estimation in DynareOBC is essentially broken.

**Comment by lulu-mao** [comment]
hi Tholden, thanks for replying.  so is there any way to do the historical decomposition considering the max and min constraints？ I saw there are three mod files automatically produced after running DynareOBC, named dynareOBCTemp1,dynareOBCTemp2 and dynareOBCTemp3. I was wondering does it make sense if I added historical decomposition code (like the following code ) at dynareOBCTemp3.mod file and run Dynare instead of dynareOBC to get the historical decomposition that I wanted？Really thanks for the helping.
varobs qh pi R yh;
calib_smoother(datafile=firm_loan_hs,diffuse_filter);
shock_decomposition(parameter_set=calibration,datafile=firm_loan_hs)qh yh;
